dspace-api/src/main/java/org/dspace/eperson/dao/impl/SubscriptionDAOImpl.java

```diff
@@ -87,8 +90,17 @@ public void deleteByCollectionAndEPerson(Context context, Collection collection,
 
     @Override
     public List<Subscription> findAllOrderedByEPerson(Context context) throws SQLException {
-        Criteria criteria = createCriteria(context, Subscription.class);
-        criteria.addOrder(Order.asc("eperson.id"));
-        return list(criteria);
+
+
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, Subscription.class);
+        Root<Subscription> subscriptionRoot = criteriaQuery.from(Subscription.class);
+        criteriaQuery.select(subscriptionRoot);
+
+        List<javax.persistence.criteria.Order> orderList = new LinkedList<>();
+        orderList.add(criteriaBuilder.asc(subscriptionRoot.get(Subscription_.ePerson)));
```

Comment:
Could not all of these 1- and 2-element Order lists be replaced by using the varargs form of CriteriaQuery.orderBy()?

Reply:
That's a good point, I'm fairly certain that that is possible. Personally I find this easier to read but if using the varargs is simpler/easier then I will refactor the instances of orderBy to use the vargs instead.